airlock/business_logic.py

```diff
@@ -539,20 +574,35 @@ def from_dict(cls, attrs) -> Self:
             },
         )
 
-    def approved_for_release(self):
-        """
-        A file is approved for release if it has been approved by two reviewers
+    def status(self) -> RequestFileReviewStatus:
+        """The status of RequestFile, based on mutliple reviews.
+
+        We specificially only require 2 APPROVED votes, rather than all votes
+        being APPROVED, as this allows a 3rd review to mark a file APPROVED to
+        unblock things if one of the initial reviewers is unavailable.
         """
-        return (
-            len(
-                [
-                    review
-                    for review in self.reviews.values()
-                    if review.status == UserFileReviewStatus.APPROVED
-                ]
-            )
-            >= 2
-        )
+        all_reviews = [v.status for v in self.reviews.values()]
+
+        if len(all_reviews) < 2:
+            # not enough votes yet
+            return RequestFileReviewStatus.INCOMPLETE
+
+        # if we have 2+ APPROVED reviews, we are APPROVED
+        if all_reviews.count(UserFileReviewStatus.APPROVED) >= 2:
+            return RequestFileReviewStatus.APPROVED
+
+        # do the reviews disagree?
+        if len(set(all_reviews)) > 1:
+            return RequestFileReviewStatus.CONFLICTED
+
+        # only case left is all reviews are REJECTED
+        return RequestFileReviewStatus.REJECTED
+
+    def status_for_user(self, user: User) -> UserFileReviewStatus | None:
```

Comment:
I think this method actually belongs in the previous commit (no need to change though)

Reply:
Yeah, I ended up doing lots of targeted fixups, suspect I got the wrong ones